crates/core/src/yarn_fn/function_wrapping.rs

```diff
@@ -170,6 +183,57 @@ macro_rules! yarn_fn_type {
 pub use yarn_fn_type;
 
 /// Adapted from <https://github.com/bevyengine/bevy/blob/fe852fd0adbce6856f5886d66d20d62cfc936287/crates/bevy_ecs/src/system/system_param.rs#L1370>
+#[cfg(feature = "bevy")]
+mod bevy_functions {
+    use super::*;
+    use bevy::ecs::system::SystemId;
+    use bevy::prelude::*;
+    use std::collections::VecDeque;
+
+    macro_rules! impl_yarn_fn_tuple_bevy {
+        ($($yarn_param: ident),*) => {
+            #[allow(non_snake_case, unused_parens)]
+            impl<'a, Output, $($yarn_param),*> YarnFn<(($($yarn_param,)*), Output)> for SystemId<In<($($yarn_param),*)>, Output>
+            where
+                $($yarn_param: TryFrom<YarnValue> + 'static,)*
+                Output: IntoYarnValueFromNonYarnValue + 'static,
+                {
+                    type Out = Output;
+                    #[allow(non_snake_case)]
+                    fn call(&self, input: Vec<YarnValue>, world: &mut World) -> Self::Out {
+                        #[allow(unused)]
+                        let mut input = VecDeque::from(input);
+                        $(
+                            assert!(!input.is_empty(), "Passed too few arguments to Function");
+                            let $yarn_param:$yarn_param = $yarn_param::try_from(input.pop_front().unwrap()).ok().expect("Invalid argument type");
```

Comment:
Is it possible to make this error message a bit more helpful? It's alright if not.

Reply:
Sort of. I added argument counts. And also to the similar message in the implementation of non-bevy functions. What you really want is the function name but there isn't enough context to do that here. This would need to become a fallible function so the error could be annotated by the caller, who presumably knows the name of the function they are invoking.

Reply:
I was primarily talking about the type error, so it could say e.g. "argument 2 has the wrong type".
But alright, that is not that useful without the function name. Let's leave that for another PR then :)